SU2_CFD/src/solvers/CHeatSolver.cpp

```diff
@@ -420,23 +352,23 @@ void CHeatSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_containe
 
   su2double *V_i, *V_j, Temp_i, Temp_i_Corrected, Temp_j, Temp_j_Corrected, **Gradient_i, **Gradient_j, Project_Grad_i, Project_Grad_j,
             **Temp_i_Grad, **Temp_j_Grad, Project_Temp_i_Grad, Project_Temp_j_Grad;
-  unsigned short iDim, iVar;
-  unsigned long iEdge, iPoint, jPoint;
-  bool flow = ((config->GetKind_Solver() == INC_NAVIER_STOKES)
-               || (config->GetKind_Solver() == INC_RANS)
-               || (config->GetKind_Solver() == DISC_ADJ_INC_NAVIER_STOKES)
-               || (config->GetKind_Solver() == DISC_ADJ_INC_RANS));
-  bool muscl = (config->GetMUSCL_Heat());
+
+  vector<su2double> Vector_i(nDim, 0.0);
+  vector<su2double> Vector_j(nDim, 0.0);
```

Comment:
nope, static of size MAXNDIM.

Reply:
As you see, I wanted to be faithful ⛪ ...but the devil himself sent `maybe-unitialized`-Warnings to derail my train of good intentions. 

(I wanted a 1-line solution to allocate and initialize with sth and I wasn't aware that `<static array>= {x}` not only sets the first value to x but actually sets all other elements to zero. Had I known that back in the day, i would have not sinned)